File: lr2/ConsoleApp2/UnitTestProject1/UnitTest1.cs
using ClassLibrary1;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;

namespace UnitTestProject1
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void minElemCAB_Test_C1()
        {
            //arrange
            double a = 1, b = 2, c = 0;
            int expected = 0;

            // act
            var result = Class1.findMin(a, b, c);

            // assert
            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void minElemABC_Test_C1()
        {
            //arrange
            double a = 1, b = 2, c = 3;
            int expected = 1; 

            // act
            var result = Class1.findMin(a, b, c);

            // assert
            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void minElemCBA_Test_C1()
        {
            //arrange
            double a = 2, b = 1, c = 0;
            int expected = 0;

            // act
            var result = Class1.findMin(a, b, c);

            // assert
            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void minElemBAC_Test_C1()
        {
            //arrange
            double a = 2, b = 1, c = 3;
            int expected = 1;

            // act
            var result = Class1.findMin(a, b, c);

            // assert
            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void findSumElements_Test_C1()
        {
            //arrange
            double[,] A = { { 1, 2 } };
            int expected = 1; 

            // act
            var result = Class1.findSumElements(A);

            // assert
            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void maxElem_Test_C1()
        {
            // arrange
            double[,] A = { { 1, 2 }, { 3, 4}};

            int expected = 4; 

            // act
            var result = Class1.maxElem(A);

            // assert
            Assert.AreEqual(expected, result);
        }
    }
}

File: lr3/ConsoleApplication1/UnitTest1/UnitTest1.cpp
#include "pch.h"
#include "CppUnitTest.h"
#include "../ConsoleApplication1/Class.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace UnitTest1
{
	TEST_CLASS(UnitTest1)
	{
	public:
		
		TEST_METHOD(getOddRankAndReverseAIsZero_Test_C2)
		{
			//arrange
			int a = 0;
			int expected = 0;

			// act
			int result = Class::getOddRankAndReverse(a);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getOddRankAndReverseAIsNotZero_Test_C2)
		{
			//arrange
			int a = 1;
			int expected = 1;

			// act
			int result = Class::getOddRankAndReverse(a);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getEvenRankWithMaxEvenValueAIsZero_Test_C2)
		{
			//arrange
			int a = 0;
			int expected = 0;

			// act
			int result = Class::getEvenRankWithMaxEvenValue(a);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getEvenRankWithMaxEvenValueAIsNotZeroWithoutEvenValues_Test_C2)
		{
			//arrange
			int a = 10;
			int expected = 0;

			// act
			int result = Class::getEvenRankWithMaxEvenValue(a);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getEvenRankWithMaxEvenValueAIsNotZeroWithEvenValue_Test_C2)
		{
			//arrange
			int a = 20;
			int expected = 2;

			// act
			int result = Class::getEvenRankWithMaxEvenValue(a);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getCycleRightShiftTwoDigitNumberWithZeroShirt_Test_C2) {
			//arrange
			int a = 12;
			int shift = 0;
			int expected = 12;

			// act
			int result = Class::getCycleRightShiftNumber(a, shift);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getCycleRightShiftTwoDigitNumberWithOneShirt_Test_C2) {
			//arrange
			int a = 12;
			int shift = 1;
			int expected = 21;

			// act
			int result = Class::getCycleRightShiftNumber(a, shift);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getCycleRightShiftOneDigitNumberWithZeroShirt_Test_C2) {
			//arrange
			int a = 9;
			int shift = 0;
			int expected = 9;

			// act
			int result = Class::getCycleRightShiftNumber(a, shift);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(getCycleRightShiftOneDigitNumberWithOneShirt_Test_C2) {
			//arrange
			int a = 9;
			int shift = 1;
			int expected = 9;

			// act
			int result = Class::getCycleRightShiftNumber(a, shift);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(sumEvenNumbers_Test_C2_0) {
			//arrange
			double expected = 0;
			double** M = new double*[0];

			// act
			double result = Class::sumEvenNumbers(M, 0);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(sumEvenNumbers_Test_C2_1) {
			//arrange
			double expected = 0;

			int size = 1;
			double** M = new double* [size];
			for (int i = 0; i < size; i++) {
				M[i] = new double[size];
				for (int j = 0; j < size; j++)
					M[i][j] = 1;
			}

			// act
			double result = Class::sumEvenNumbers(M, size);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(sumEvenNumbers_Test_C2_21) {
			//arrange
			double expected = 0;
			int size = 2;
			double** M = new double* [size];
			for (int i = 0; i < size; i++) {
				M[i] = new double[size];
				for (int j = 0; j < size; j++)
					M[i][j] = 1;
			}

			// act
			double result = Class::sumEvenNumbers(M, size);

			// assert
			Assert::AreEqual(expected, result);
		}

		TEST_METHOD(sumEvenNumbers_Test_C2_22) {
			//arrange
			double expected = 2;

			int size = 2;
			double** M = new double* [size];
			for (int i = 0; i < size; i++) {
				M[i] = new double[size];
				for (int j = 0; j < size; j++)
					M[i][j] = 2;
			}

			// act
			double result = Class::sumEvenNumbers(M, size);

			// assert
			Assert::AreEqual(expected, result);
		}
	};
}
